--- a/modules/dream2nix/WIP-python-pdm/lib.nix
+++ b/modules/dream2nix/WIP-python-pdm/lib.nix
@@ -176,9 +176,11 @@
   parsePackage = environ: item: let
+    isPathPackage = item ? path;
+    files = if isPathPackage then [{file = item.path;}] else item.files;
-    sources = sourcesToAttrs item.files;
+    sources = sourcesToAttrs files;
     compatibleSources =
       lib.filterAttrs
       (
         filename: source:
-          isUsableFilename {inherit environ filename;}
+          if isPathPackage then true else isUsableFilename {inherit environ filename;}
       )
       sources;
